src/shared/sam/cli/samCliInvokerUtils.ts

```diff
@@ -55,6 +55,42 @@ export function logAndThrowIfUnexpectedExitCode(processResult: ChildProcessResul
             message = processResult.error.message
         }
     }
+    const usefulErrors = collectAcceptedErrorMessages(processResult.stderr).join('\n')
+    throw makeUnexpectedExitCodeError(message ?? usefulErrors)
+}
+
+/**
+ * Collect known errors messages from a sam error message
+ * that may have multiple errors in one message.
+ * @param errors A string that can have multiple error messages
+ */
+export function collectAcceptedErrorMessages(errorMessage: string): string[] {
+    const errors = errorMessage.split('\n')
+    const shouldCollectFuncs = [_startsWithEscapeSequence, _startsWithError]
+    return errors.filter(error => {
+        return shouldCollectFuncs.some(shouldCollect => {
+            return shouldCollect(error)
+        })
+    })
+}
+
+/**
+ * Returns true if text starts with an escape
+ * sequence with one of the accepted sequences.
+ */
+function _startsWithEscapeSequence(text: string): boolean {
+    const escapeInDecimal = 27
+    if (text.charCodeAt(0) !== escapeInDecimal) {
+        return false
+    }
+
+    const remainingText = text.substring(1)
+    const acceptedSequences = ['[33m']
```

Comment:
ANSI escape codes are somewhat obscure so a comment or variable is helpful:
```suggestion
    // ANSI Escape Sequences
    // ref: https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797#8-16-colors
    const acceptedSequences = [
        '[33m' // Yellow Foreground
     ]
```
Comments are fine here IMO but you can always extract into a variable for 'self-documenting' code:
```ts
const yellowForeground =  '[33m' 
const acceptedSequences = [yellowForeground]
```

Reply:
I like the last example, updated to this.